### DISCUSSION TITLE
`match_ID` returned by `DependencyMatcher._call__` method should be type `str` not `int`

### DISCUSSION THREAD

**Discussion by omarzd** [explosion/spaCy#7391] [New Features & Project Ideas]
The method [`DependencyMatcher._call__`](https://spacy.io/api/dependencymatcher#call) returns `match_ID` as type `int` even though `match_ID` is type `str` everywhere else including when it is set by the user. I think for the sake of consistency and usability, this should be fixed.

**Comment by adrianeboyd** [comment]
Hi, spacy stores most strings internally as ints for efficiency, so the matchers (`Matcher`, `PhraseMatcher`, and `DependencyMatcher`) also do this for the match IDs. As far as I know they've always returned matches in this format.

I can totally understand why you'd prefer strings instead, but we want to keep these kinds of APIs the same as much as possible and we want to keep it as efficient as possible internally, too.

For `Matcher`, there's the new `as_spans` option which is a little more user-friendly, but there's no good alternative format like `Span` for storing non-contiguous dependency match results.
